server/src/main/java/org/apache/iotdb/db/tools/TsFileSelfCheckTool.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.db.tools;
+
+import org.apache.iotdb.tsfile.exception.TsFileStatisticsMistakesException;
+import org.apache.iotdb.tsfile.file.metadata.MetadataIndexEntry;
+import org.apache.iotdb.tsfile.file.metadata.MetadataIndexNode;
+import org.apache.iotdb.tsfile.file.metadata.TimeseriesMetadata;
+import org.apache.iotdb.tsfile.file.metadata.enums.MetadataIndexNodeType;
+import org.apache.iotdb.tsfile.read.TsFileSequenceReader;
+import org.apache.iotdb.tsfile.read.common.Path;
+import org.apache.iotdb.tsfile.utils.Pair;
+
+import java.io.IOException;
+import java.nio.BufferOverflowException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class TsFileSelfCheckTool {
+
+  public long check(String filename, boolean fastFinish)
+      throws IOException, TsFileStatisticsMistakesException {
+    System.out.println("file path: " + filename);
+
+    TsFileSelfCheckToolReader reader = new TsFileSelfCheckToolReader(filename);
+    Map<Long, Pair<Path, TimeseriesMetadata>> timeseriesMetadataMap = null;
+    try {
+      timeseriesMetadataMap = reader.getAllTimeseriesMetadataWithOffset();
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
Use logger.error rather than print stack trace

Reply:
I've changed printStackTrace to logger.